lib/rails_semantic_logger/extensions/active_support/logger.rb

```diff
@@ -4,7 +4,7 @@ module ActiveSupport
   # More hacks to try and stop Rails from being it's own worst enemy.
   class Logger
     class << self
-      undef :logger_outputs_to?, :broadcast
+      undef :logger_outputs_to?
```

Comment:
We still need to `undef :broadcast` for the previous Rails versions to prevent a warning, please `undef_method(:broadcast) if method_defined?(:broadcast)` like [here](https://github.com/reidmorrison/rails_semantic_logger/blob/master/lib/rails_semantic_logger/extensions/rails/server.rb#L8)

Also should be tested, please add 7.1 to CI. I guess [here](https://github.com/reidmorrison/rails_semantic_logger/blob/master/.github/workflows/ci.yml#L41-L42) Rails-7.0 could be replaced by 7.1

Reply:
...and CI would probably fail, because we're undefining methods only to override it later, since it's not defined anymore and replaced with other API -- some functionality would be broken.

Reply:
quick & dirty commit just to run CI https://github.com/ojab/rails_semantic_logger/tree/rails-7.1, it's failing in unexpected place, but indeed failing https://github.com/ojab/rails_semantic_logger/actions/runs/6461173004/job/17540325867

Reply:
added 7.1 to the matrix and implemented @camsteffen's solution above

Reply:
> Made a PR to fix it [rails/rails#49563](https://github.com/rails/rails/pull/49563), will wait a couple of days for feedback and [if accepted] would try to make a PR to `rails_semantic_logger`, _probably_ with out-of-the-box monkeypatch.

looks like a patch was merged into Rails
https://github.com/rails/rails/pull/49621

Reply:
running 

```ruby
  gem "rails_semantic_logger", git: "https://github.com/Amnesthesia/rails_semantic_logger.git", branch: "fix/rails7.1"
  gem "rails", git: "https://github.com/rails/rails.git", branch: "7-1-stable"
```
all is working
```
❯ DEPENDENCIES_NEXT=1 rc
Loading development environment (Rails 7.1.1)
 development  irb(main):001> 
```